cmd/minikube/cmd/start.go

```diff
@@ -1185,6 +1191,31 @@ func validateKubernetesVersion(old *config.ClusterConfig) {
 	}
 }
 
+// validateBaseImage checks that --base-image is not passed if the drive being in use is KIC (docker/podman)
+// if so, the function exits the process
+func validateBaseImage(imageFlag *pflag.Flag, driver string) {
+	if !validBaseImageFlag(imageFlag.Changed, driver) {
+		exit.Message(reason.Usage,
+			"flag --{{.imgFlag}} is not available for driver '{{.driver}}'. Did you mean to use '{{.docker}}' or '{{.podman}}' driver instead?\n"+
+				"Please use --{{.isoFlag}} flag to configure VM based drivers",
+			out.V{
+				"imgFlag": imageFlag.Name,
+				"driver":  driver,
+				"image":   imageFlag.Value,
+				"docker":  registry.Docker,
+				"podman":  registry.Podman,
+				"isoFlag": isoURL,
+			},
+		)
+	}
+}
+func validBaseImageFlag(baseImageFlagSet bool, driver string) bool {
+	if baseImageFlagSet {
+		return registry.IsKIC(driver)
```

Comment:
to make sure we cover Baremetal (none driver)
it is better we check using IsVM here https://github.com/medyagh/minikube/blob/107eb57a110a4a650fc125401719b51e6eccf111/pkg/minikube/driver/driver.go#L105
```
driver.IsVM(drver)
```